lnrpc/invoicesrpc/addinvoice_test.go

```diff
@@ -0,0 +1,575 @@
+			TimeLockDelta:             2000,
+		}
+
+		// Create a hop hint based on privateChan2Policy.
+		privateChannel2Hint = zpay32.HopHint{
+			NodeID:      privateChannel2.RemotePubkey,
+			ChannelID:   private2ShortID.ToUint64(),
+			FeeBaseMSat: uint32(privateChan2Policy.FeeBaseMSat),
+			FeeProportionalMillionths: uint32(
+				privateChan2Policy.FeeProportionalMillionths,
+			),
+			CLTVExpiryDelta: privateChan2Policy.TimeLockDelta,
+		}
+
+		// Create a third private channel that we'll use for hints.
+		private3ShortID = lnwire.NewShortChanIDFromInt(3)
+		privateChannel3 = &HopHintInfo{
+			IsPublic: false,
+			FundingOutpoint: wire.OutPoint{
+				Index: 3,
+			},
+			RemotePubkey:   pubkey,
+			RemoteBalance:  100,
+			ShortChannelID: private3ShortID,
+		}
+		private3ChannelID = lnwire.NewChanIDFromOutPoint(
+			&privateChannel3.FundingOutpoint,
+		)
+
+		// Create a edge policy for private channel 1.
+		privateChan3Policy = &channeldb.ChannelEdgePolicy{
+			FeeBaseMSat:               30,
+			FeeProportionalMillionths: 300,
+			TimeLockDelta:             3000,
+		}
+
+		// Create a hop hint based on privateChan2Policy.
+		privateChannel3Hint = zpay32.HopHint{
+			NodeID:      privateChannel3.RemotePubkey,
+			ChannelID:   private3ShortID.ToUint64(),
+			FeeBaseMSat: uint32(privateChan3Policy.FeeBaseMSat),
+			FeeProportionalMillionths: uint32(
+				privateChan3Policy.FeeProportionalMillionths,
+			),
+			CLTVExpiryDelta: privateChan3Policy.TimeLockDelta,
+		}
+	)
+
+	// We can't copy in the above var decls, so we copy in our pubkey here.
+	var peer [33]byte
+	copy(peer[:], compressed)
+
+	var (
+		// We pick our policy based on which node (1 or 2) the remote
+		// peer is. Here we create two different sets of edge
+		// information. One where our peer is node 1, the other where
+		// our peer is edge 2. This ensures that we always pick the
+		// right edge policy for our hint.
+		infoNode1 = &channeldb.ChannelEdgeInfo{
+			NodeKey1Bytes: peer,
+		}
+
+		infoNode2 = &channeldb.ChannelEdgeInfo{
+			NodeKey1Bytes: [33]byte{9, 9, 9},
+			NodeKey2Bytes: peer,
+		}
+
+		// setMockChannelUsed preps our mock for the case where we
+		// want our private channel to be used for a hop hint.
+		setMockChannelUsed = func(h *hopHintsConfigMock,
+			channelID lnwire.ChannelID,
+			shortID lnwire.ShortChannelID,
+			policy *channeldb.ChannelEdgePolicy) {
+
+			// Return channel is active so we'll consider it for
+			// hints.
+			h.Mock.On(
+				"IsChannelActive", channelID,
+			).Once().Return(true)
+
+			// Return public node = true so that we'll consider
+			// this node for our hop hints.
+			h.Mock.On(
+				"IsPublicNode", peer,
+			).Once().Return(true, nil)
+
+			// When it gets time to find an edge policy for this
+			// node, fail it. We won't use it as a hop hint.
+			h.Mock.On(
+				"FetchChannelEdgesByID",
+				shortID.ToUint64(),
+			).Once().Return(
+				infoNode1, policy, otherChanPolicy, nil,
+			)
+		}
+	)
+
+	tests := []struct {
```

Comment:
In case you need it in the future what I do is declare the slice of tests outside the function so you earn some extra space (80 characters can be quite limiting when you are doing table driven tests)